sdk/evm/tbrv3/contract.ts

```diff
@@ -118,27 +133,79 @@ export interface Transfer {
 
 export class Tbrv3 {
 
-  readonly address: string;
-
-  constructor(public readonly provider: ethers.Provider, public readonly network: NetworkMain, address: string = Tbrv3.addresses[network]) {
+  constructor(
+    public readonly provider: ethers.Provider,
+    public readonly network: NetworkMain,
+    public readonly address: string = Tbrv3.addresses[network],
+    private readonly gasToken: string,
+  ) {
     this.address = address;
+
+  }
+
+  static connect(
+    provider: ethers.Provider,
+    network: NetworkMain,
+    chain: Chain,
+    gasToken?: string,
+    address?: string,
+  ) {
+    if (address === undefined && this.addresses[network] === "TBD") {
+      throw new Error(`Tbrv3 address needs to be provided for network ${network}`);
+    }
+
+    const defaultGasToken = this.gasTokens[network][chain];
+    if (gasToken === undefined) {
+      if (defaultGasToken === undefined) {
+        throw new Error(`Gas token address needs to be provided for network ${network} and chain ${chain}`);
+      }
+      gasToken = defaultGasToken;
+    } else if (defaultGasToken !== undefined && gasToken.toLowerCase() !== defaultGasToken?.toLowerCase()) {
+      throw new Error(`Unexpected gas token address ${gasToken} for network ${network} and chain ${chain}`);
+    }
+
+    return new Tbrv3(provider, network, address, gasToken);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrlStatic(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc, undefined, {staticNetwork: true}), network, chain, gasToken, address);
+  }
+
+  /**
+   * @deprecated This method will be removed soon. The sdk won't use ethers directly anymore.
+   */
+  static fromRpcUrl(rpc: string, network: NetworkMain, chain: Chain, gasToken?: string, address?: string): Tbrv3 {
+    return this.connect(new ethers.JsonRpcProvider(rpc), network, chain, gasToken, address);
   }
 
   static readonly addresses: Record<NetworkMain, string> = {
     Mainnet: "TBD",
     Testnet: "TBD",
   }
 
-  transferWithRelay(...transfers: Transfer[]): TbrPartialTx {
+  // TODO: fill these out
+  static readonly gasTokens: Record<NetworkMain, Partial<Record<Chain, string>>> = {
+    Mainnet: {},
+    Testnet: {},
+  }
```

Comment:
Should come from the SDKv2.

Reply:
Are these already defined there or should we add them?

Reply:
Yep, if you look [in the tokens constants of the SDKv2](https://github.com/wormhole-foundation/wormhole-sdk-ts/blob/fa4ba4bc349a7caada809f209090d79a3c5962fe/core/base/src/constants/tokens/mainnet.ts#L15-L16) you'll find that the gas token have an `address: 'native'` field and also point to their wrapped representations.

Reply:
Nice, I'm going to hook that up now then.

Reply:
Done.